src/main/java/network/brightspots/rcv/ContestConfig.java

```diff
@@ -518,6 +517,12 @@ private void validateCvrFileSources() {
             isValid = false;
             Logger.log(Level.SEVERE, "CDF files must be tabulated individually.");
           }
+          if (!source.getFilePath().endsWith(".xml") && !source.getFilePath().endsWith(".json")) {
+            Logger.log(Level.SEVERE,
+                "Unexpected file extension: %s.  CDF source files must be .xml or .json",
+                source.getFilePath());
+            isValid = false;
+          }
```

Comment:
Actually, could you please move this to `passesBasicCvrSourceValidation`? That way it triggers when attempting to add a CVR in the GUI, instead of just when attempting to tabulate.